### PR TITLE
[Compute] `az restore-point`: Add new command group `restore-point` and `restore-point collection` to support RestorePoints operations

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

RestorePoints feature https://github.com/Azure/azure-cli/issues/18844
**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change.
[Component Name 2] `az command b`: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

**Comment by zhoxing-ms** [comment]
Because this PR needs to wait for the update of the new Python SDK before merging, so make it with tag `do not merge` first

File: src/azure-cli/azure/cli/command_modules/vm/_help.py
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:3496]: Please use the active voice in the help description
```suggestion
    short-summary: "Get the list of restore point collections in the subscription. Use nextLink property in the \
response to get the next page of restore point collections. Do this till nextLink is not null to fetch all the restore \
point collections."
```

File: src/azure-cli/azure/cli/command_modules/vm/_help.py
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_help.py:3452]:
```suggestion
    short-summary: "Get the list of restore point collections in a resource group."
    examples:
      - name: Get the list of restore point collections in a resource group.
```

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
